seatunnel-translation/seatunnel-translation-base/src/main/java/org/apache/seatunnel/translation/serialization/RowConverter.java

```diff
@@ -80,8 +82,9 @@ protected boolean validate(Object field, SeaTunnelDataType<?> dataType) {
             case STRING:
             case DECIMAL:
             case BYTES:
-            case ARRAY:
                 return dataType.getTypeClass() == field.getClass();
+            case ARRAY:
+                return ClassUtils.wrapperToPrimitive(dataType.getTypeClass().getComponentType()) == field.getClass().getComponentType();
```

Comment:
If source produce References Array, seem like the logic will return false.

Reply:
![image](https://user-images.githubusercontent.com/32387433/192952067-f8ba7645-494e-4822-b0dc-69c046fe2cae.png)
Check this

Reply:
I mean you should check both primitive and references type.